pkg/services/ngalert/eval/eval.go

```diff
@@ -247,6 +247,7 @@ func getExprRequest(ctx EvaluationContext, data []models.AlertQuery, dsCacheServ
 	req := &expr.Request{
 		OrgId:   ctx.User.OrgID,
 		Headers: buildDatasourceHeaders(ctx.Ctx),
+		User:    ctx.User,
```

Comment:
How do you use this User downstream? Asking because the scheduler that executes this method uses a fake user 
https://github.com/grafana/grafana/blob/9eb10bee1f2d249b746c8011f7446b10eec94c98/pkg/services/ngalert/schedule/schedule.go#L574-L589

Reply:
I have seen it only used as purely informational - and the plugin SDK docs highlights this detail about it potentially coming from an alerting context https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/common.go#L137-L141 and thus not to be relied on.